src/stores/new-sites-slice.ts

```diff
@@ -0,0 +1,70 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import { store } from 'src/stores/index';
+
+interface NewSitesState {
+	processingSiteIds: string[];
+}
+
+const initialState: NewSitesState = {
+	processingSiteIds: [],
+};
+
+const newSitesSlice = createSlice( {
+	name: 'newSites',
+	initialState,
+	reducers: {
+		addProcessingSites: ( state, action: PayloadAction< string[] > ) => {
+			state.processingSiteIds = [ ...new Set( [ ...state.processingSiteIds, ...action.payload ] ) ];
+		},
+		removeProcessingSites: ( state, action: PayloadAction< string[] > ) => {
+			state.processingSiteIds = state.processingSiteIds.filter(
+				( id ) => ! action.payload.includes( id )
+			);
+		},
+	},
+} );
+
+export const { addProcessingSites, removeProcessingSites } = newSitesSlice.actions;
```

Comment:
```suggestion
export const newSitesActions = newSitesSlice.actions;
```
Nit, but it'd be nice to stay consistent with other slices and export the actions inside a "namespace".